### PR TITLE
refactor: cleanup

### PR DESCRIPTION
**Pull request by SrIzan10**
This PR features basic refactoring and cleaning up.
Users may receive some errors if they were using **SernEmitter** directly. This class has been removed

### CODE REVIEW

**Comment by jacoobes** [comment]
Removed the localization feature of this branch. this has turned into a refactor branch.
Localization is moved to sern/tools

**Review by jacoobes — APPROVED** [review]

**Comment by jacoobes** [comment]
By at least tomorrow, this might be a patch I gotta review it again

Vào Th 5, 8 thg 2, 2024 lúc 12:31 CH SrIzan10 ***@***.***> đã
viết:

File: src/core/ioc/container.ts [hunk 1]
```diff
@@ -33,19 +35,25 @@ export class CoreContainer<T extends Partial<Dependencies>> extends Container<T,
...
 
     override async disposeAll() {
-        
         const otherDisposables = Object
             .entries(this._context)
             .flatMap(([key, value]) => 
                 'dispose' in value ? [key] : []);
-
-        for(const key of otherDisposables) {
+        otherDisposables.forEach(key => {
```
Comment (SrIzan10) [inline src/core/ioc/container.ts:48]: why did you refactor this to be a forEach instead of a for loop?

File: src/core/ioc/index.ts [hunk 2]
```diff
@@ -1,2 +1,2 @@
 export { makeDependencies } from './base';
-export { Service, Services, single, transient } from './dependency-injection';
+export { Service, Services, single, transient  } from './dependency-injection';
```
Comment (SrIzan10) [inline src/core/ioc/index.ts:2]: woah that space there!

**Review by SrIzan10 — COMMENTED** [review]
I can't request changes on my own PR, but take a look at this.

File: src/core/ioc/container.ts (hunk 1 above)
Comment (jacoobes) [reply src/core/ioc/container.ts:48]: A little easier to read

**Comment by jacoobes** [comment]
@SrIzan10 rereview, I think everything will be alright

**Pull request merged**